clarity/src/vm/errors.rs

```diff
@@ -144,6 +144,7 @@ pub enum WasmError {
     ExpectedReturnValue,
     InvalidIndicator(i32),
     Runtime(wasmtime::Error),
+    UnhandledType,
```

Comment:
This error should instead be `InvalidType`. We don't want an error for a type that is not handled yet, just an error for a type that should not be allowed where it was found.

Reply:
Or you could use the existing `InvalidNoTypeInValue` and add `InvalidListUnionTypeInValue`.